ejb3/src/main/java/org/jboss/as/ejb3/component/EJBComponentDescription.java

```diff
@@ -682,12 +682,22 @@ public void setDefaultSecurityDomain(final String defaultSecurityDomain) {
         this.defaultSecurityDomain = defaultSecurityDomain;
     }
 
-    public void setKnownSecurityDomainPredicate(final Predicate<String> knownSecurityDomain) {
+    public void setKnownSecurityDomainPredicate(final Function<String, ApplicationSecurityDomainConfig> knownSecurityDomain) {
```

Comment:
Just a minor nit, this method should probably be renamed to "setKnownSecurityDomainFunction" now.